torch/distributed/_tensor/api.py

```diff
@@ -343,6 +343,14 @@ def from_local(
         # set default placements to replicated if not specified
         if placements is None:
             placements = [Replicate() for _ in range(device_mesh.ndim)]
+        else:
+            placements = list(placements)
+            for idx, placement in enumerate(placements):
+                # normalize shard dim to be positive
+                if placement.is_shard():
```

Comment:
@wanchaol  Should we converge to using `placement.is_shard()` or to using `isinstance(placement, Shard)`? The former calls the latter but allows for passing a `dim` arg to further check against, and the latter avoids having to use `cast(Shard, placement)`.

It seems like `is_shard()` is a higher level construct and should be preferred, but I wanted to check.

Reply:
that's exactly the trade off you pointed out lol, I would like to use the former uniformly, but mypy can't recognize it as a result there're many redundant cast needed if we switch all callsite to that..

I think maybe we can do this:
- isinstance(placement, Shard) preferred if do simple type check
- is_shard(`dim`) where `dim` become non-optional, so that this API only used as a util to check if the placement is shard on a certain tensor dim